internal/controller/bootstrap/controlplane_bootstrap_controller.go

```diff
@@ -155,6 +156,25 @@ func (c *ControlPlaneController) Reconcile(ctx context.Context, req ctrl.Request
 	)
 
 	if config.Spec.K0s != nil {
+
+		minVersionForETCDName := version.MustParse("v1.31.1+k0s.0")
+		currentKCPVersion := version.MustParse(config.Spec.Version)
```

Comment:
It's shouldn't happen, but just to be safe, let's return an error instead of panic here